### PR TITLE
Messaging #87

### PR DESCRIPTION
**Pull request by eric-silva-61**
### What this PR does (required):
- Added MessagingContext
- Context transforms API data to match what UI expects
- Adjusted BE API to return all conversation and messages at once
- Updates to UI to use new context

### Screenshots / Videos (required):
image

### Any information needed to test this feature (required):
- Go to messaging tab using the demo login

### Any issues with the current functionality (optional):
-none

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -30,25 +31,27 @@ function App(): JSX.Element {
...
+                  <MessagingProvider>
+                    {/* Renders navbar for all pages */}
+                    <Route path="/" component={NavBar} />
+                    <Route exact path="/login" component={Login} />
+                    <Route exact path="/signup" component={Signup} />
+                    <ProtectedRoute exact path="/file-upload/:id" component={DesignSubmit} />
+                    <ProtectedRoute exact path="/setting" component={ProfileSetting} />
+                    {/* Replace component with Discovery Page */}
+                    <Route exact path="/discovery" component={Discovery} />
+                    {/* Replace the components once created */}
```
Comment (rajivtitus) [inline client/src/App.tsx]: As we won't be adding any more new components, please feel free to remove the comments here. Thanks!

File: client/src/context/useMessagingContext.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  conversations: Conversation[];
+}
+
+export const MessagingContext = createContext<IMessagingContext>({
+  conversations: [],
+});
+
+export const MessagingProvider: FC = ({ children }): JSX.Element => {
+  const { loggedInUser } = useContext(AuthContext);
+  const [conversations, setConversations] = useState<Conversation[]>([]);
```
Comment (rajivtitus) [inline client/src/context/useMessagingContext.tsx:43]: How are we planning on updating and displaying the messages within a conversation? Also, how are we going to differentiate which conversation is the active one?

File: client/src/App.tsx (hunk 1 above)
Comment (eric-silva-61) [reply client/src/App.tsx]: Done

File: client/src/context/useMessagingContext.tsx (hunk 2 above)
Comment (eric-silva-61) [reply client/src/context/useMessagingContext.tsx:43]: Added add message functionality to context, assuming that with the socket io integration the backend would update the database.  Added indicator for active conversation.
image

**Comment by eric-silva-61** [comment]
Also added in new conversation functionality.  

- If conversation already exists returns id
- if new conversation, creates it on server and returns new id
- added route (/messages/:id) to handle going to messages and loading correct conversation

File: client/src/context/useMessagingContext.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  newConversation: (to: string) => void;
+}
+
+export const MessagingContext = createContext<IMessagingContext>({
+  conversations: [],
+  addMessage: () => null,
+  newConversation: () => null,
+});
+
+// const newConversation = (conversations: APIConversation[], loggedInUserEmail: string): Conversation[] =>
```
Comment (rajivtitus) [inline client/src/context/useMessagingContext.tsx:19]: Please remove commented code if we are no longer using it

**Review by rajivtitus — APPROVED** [review]
LGTM! Thanks for making requested changes

**Pull request merged**